classes/Kohana/Response.php

```diff
@@ -556,10 +556,13 @@ public function send_file($filename, $download = NULL, array $options = NULL)
 				if (is_object(Kohana::$log))
 				{
 					// Add this exception to the log
-					Kohana::$log->add(Log::ERROR, $error);
+					Kohana::$log->error($error);
 
-					// Make sure the logs are written
-					Kohana::$log->write();
+					if (Kohana::$log instanceof Kohana_Logger)
```

Comment:
@Yahasana kindly note that this `if` test is to insure that the `$log` can `write()`.